### PR TITLE
Bump `nginx-ingress-controller` addon image to v1.3.0 for 1.22+ shoots

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
For shoot addons, we're using a pretty old nginx version which has lot of CVEs.
We can't directly update to the latest version because of "migration to leases".
We need to use `v1.3.0` for one release and then we can go for the latest image.
The required RBAC for leases are updated. Similar to https://github.com/gardener/gardener/pull/6356
See compatibility matrix: https://github.com/kubernetes/ingress-nginx#supported-versions-table

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other dependency
Shoot addon `nginx-ingress-controller` image is updated to `v1.3.0` for `v1.22+` shoots.
```

### CODE REVIEW

**Comment by shafeeqes** [comment]
Oops!
/milestone v1.71

File: pkg/operation/botanist/component/nginxingressshoot/nginxingress.go [hunk 1]
```diff
@@ -489,7 +505,7 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 				{
 					APIGroups:     []string{""},
 					Resources:     []string{"configmaps"},
-					ResourceNames: []string{"ingress-controller-leader-nginx"},
+					ResourceNames: []string{"ingress-controller-leader"},
```
Comment (acumino) [inline pkg/operation/botanist/component/nginxingressshoot/nginxingress.go:508]: Changing the resource name will cause the existing running setup to break, and leader election will be required to be performed again.
```suggestion
					ResourceNames: []string{"ingress-controller-leader-nginx"},
```

Comment (shafeeqes) [reply pkg/operation/botanist/component/nginxingressshoot/nginxingress.go:508]: This is the name of the configmap created now also, we pass this here, https://github.com/gardener/gardener/blob/12518e2da78817ace8178c1afca928d15b9cc6e3/pkg/operation/botanist/component/nginxingressshoot/nginxingress.go#L719

Comment (acumino) [reply pkg/operation/botanist/component/nginxingressshoot/nginxingress.go:508]: /approve

**Pull request merged**